scripts/stackstorm.js

```diff
@@ -258,8 +258,10 @@ module.exports = function(robot) {
         robot.logger.info(command_factory.st2_hubot_commands.length + ' commands are loaded');
       })
       .catch(function (err) {
-        var error_msg = 'Failed to retrieve commands from "%s": %s';
-        robot.logger.error(util.format(error_msg, env.ST2_API_URL, err.message));
+        robot.logger.error(util.format('Failed to retrieve commands from "%s": %s', env.ST2_API_URL, err.message));
+        if (err.status === 401 || err.message.includes('Unauthorized')) {
+          throw err;
```

Comment:
Not happy with this change either right now as it breaks any attempt of `hubot-stackstorm` to recover from failure and re-request st2 token as it happens normally.

Here is how it works with current PR:
```
INFO Requesting a token...
INFO Loading commands....
ERROR Failed to retrieve commands from "https://localhost/api": Unauthorized - Token has expired.
ERROR {"name":"APIError","status":401,"message":"Unauthorized - Token has expired."}
INFO Hubot will shut down ...
INFO Disconnected from Slack RTM
INFO Exiting...
```

In this case request-token loop clashes in timing with the reload commands loop and so token was requested, but not received yet. At that exact moment reload commands loop happens that fails with `401 Token has expired` because we hadn't enough time to receive the updated token.

To address that I think we need to modify the loop in `st2client.js` here: https://github.com/StackStorm/st2client.js/blob/46562a2b4959308d82998cc309b076f6c42d0ec2/index.js#L183-L201
and re-request token `1-3s` earlier in advance. That will give enough timing window.

I think that might be one of the issues behind #157

Reply:
https://github.com/StackStorm/st2client.js/pull/69 should fix it which emits st2 client expiry token `1s` earlier, giving some time to re-authenticate that should fix some race condition cases.

Reply:
@armab I think this is resolved with StackStorm/st2client.js#69 merged.